src/flameGraph.cpp

```diff
@@ -489,23 +712,51 @@ int FlameGraph::selectFrameColor(std::string& name) {
     if (StringUtils::endsWith(name, "_[j]", 4)) {
         // Java compiled frame
         name = name.substr(0, name.length() - 4);
-        return green.getColor();
+        return palette ? green.getColor() : GREEN;
     } else if (StringUtils::endsWith(name, "_[i]", 4)) {
         // Java inlined frame
         name = name.substr(0, name.length() - 4);
-        return aqua.getColor();
+        return palette ? aqua.getColor() : AQUA;
     } else if (StringUtils::endsWith(name, "_[k]", 4)) {
         // Kernel function
         name = name.substr(0, name.length() - 4);
-        return brown.getColor();
+        return palette ?  brown.getColor() : BROWN;
     } else if (name.find("::") != std::string::npos) {
         // C++ function
-        return yellow.getColor();
+        return palette ? yellow.getColor() : YELLOW;
     } else if ((int)name.find('/') > 0 || ((int)name.find('.') > 0 && name[0] >= 'A' && name[0] <= 'Z')) {
         // Java regular method
-        return green.getColor();
+        return palette ? green.getColor() : GREEN;
     } else {
         // Other native code
-        return red.getColor();
+        return palette ? red.getColor() : RED;
```

Comment:
It looks confusing when the function returns either a color or enum value.
I've pushed the refactoring that will help to use palette names without ambiguous arguments and without extra `switch`.